src/pages/index.js

```diff
@@ -39,38 +42,183 @@ const profFormPopup = new PopupWithForm(
   ".modal_type_profile",
   handleProfileSubmit
 );
+const delCardFormPopup = new PopupWithForm(
+  ".modal_type_delete-card",
+  handleDelCardSubmit
+);
+const editAviFormPopup = new PopupWithForm(
+  ".modal_type_edit-avi",
+  handleEditAviSubmit
+);
 const profUserInfo = new UserInfo({
   userNameSelec: ".profile__name",
   descrSelec: ".profile__title",
+  aviSelec: ".profile__avatar",
+});
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: apiHeaders,
 });
+const delCardApi = new Api({
+  method: "DELETE",
+  headers: apiHeaders,
+});
+
+//INITIALLY SET USER INFO
+api
+  .getUserInfo()
+  .then((data) => {
+    const { name, about, avatar } = data;
+    profUserInfo.setUserInfo({ name, about, avatar });
+  })
+  .catch((err) => console.error(err));
 
 // POPULATE INITIAL CARDS
 export function createCard(data) {
-  const cardElement = new Card(data, "#card-template", handleCardEnlargement);
+  const cardElement = new Card(
+    data,
+    "#card-template",
+    handleCardEnlargement,
+    handleTrashClick,
+    sendLikeReq
+  );
   return cardElement.render();
 }
+api
+  .getCards()
+  .then((data) => {
+    data.reverse();
+    const initialSection = new Section(
+      { items: data, renderer: createCard },
+      ".gallery"
+    );
 
-section.renderItems();
+    initialSection.renderItems();
+  })
+  .catch((err) => console.error(err));
 
 // ENLARGE CARD IMAGE
 function handleCardEnlargement(evt) {
   enlrgImgPopup.open(evt);
 }
 
 // SUBMIT NEW CARD MODAL
-export function handleNewCardFormSubmit(newCardData) {
-  section.addItem(createCard(newCardData));
+function handleNewCardFormSubmit({ name, link }) {
+  const newCardApi = new Api({
+    method: "POST",
+    headers: apiHeaders,
+    body: JSON.stringify({
+      name,
+      link,
+    }),
+  });
 
-  newCardFormPopup.close();
-  newCardModalForm.reset();
+  newCardFormPopup.popup.querySelector(".modal__submit-button").textContent =
+    "Creating...";
 
-  newCardFormValidator.resetValidation();
+  newCardApi
+    .getCards()
+    .then((data) => {
+      const newCardSection = new Section(
+        { items: [data], renderer: createCard },
+        ".gallery"
+      );
+      newCardSection.renderItems();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      newCardFormPopup.popup.querySelector(
+        ".modal__submit-button"
+      ).textContent = "Create";
+      newCardFormPopup.close();
+      newCardModalForm.reset();
+      newCardFormValidator.resetValidation();
```

Comment:
```js

      newCardFormPopup.close();
      newCardModalForm.reset();
      newCardFormValidator.resetValidation();
```

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server)  


**Please, correct it everywhere**